### PR TITLE
swift-format: build with swift on Catalina and Linux

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

We probably don't want to merge this yet, as this depends on the outcome of Homebrew/brew#11964. However, it would be good to test this to check that it works.

### CODE REVIEW

**Comment by Bo98** [comment]
`swift-format` is a bit of an unusual one. It actually relies on the toolchain (`lib_InternalSwiftSyntaxParser`), so technically that Xcode requirement is probably a runtime one.

It looks like Catalina might still be looking at Xcode's one rather than the one in our toolchain. Could do with some `otool` debugging on this.

Ideally the syntax parser would link statically, this isn't supported though (yet): https://bugs.swift.org/browse/SR-14001

---

In general, Swift would probably be a buildtime dependency on Catalina (but runtime on Linux) - we do not ship a standard library in our Swift toolchain on macOS. This would be the one exception where it's runtime on Catalina too.

Given this one is unusual, it's worth experimenting with Swift 5.4 on Catalina on other formulae too.

**Comment by carlocab** [comment]
```
❯ otool -L swift-format
swift-format:
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1770.255.0)
        /usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
        /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1770.255.0)
        @rpath/lib_InternalSwiftSyntaxParser.dylib (compatibility version 1.0.0, current version 17013.0.0)
        @rpath/libswiftCore.dylib (compatibility version 1.0.0, current version 1200.2.41)
        @rpath/libswiftCoreFoundation.dylib (compatibility version 1.0.0, current version 1.6.0, weak)
        @rpath/libswiftCoreGraphics.dylib (compatibility version 1.0.0, current version 2.0.0, weak)
        @rpath/libswiftDarwin.dylib (compatibility version 1.0.0, current version 0.0.0)
        @rpath/libswiftDispatch.dylib (compatibility version 1.0.0, current version 4.40.2)
        @rpath/libswiftFoundation.dylib (compatibility version 1.0.0, current version 20.0.0)
        @rpath/libswiftIOKit.dylib (compatibility version 1.0.0, current version 1.0.0, weak)
        @rpath/libswiftObjectiveC.dylib (compatibility version 1.0.0, current version 1.0.0, weak)
        @rpath/libswiftXPC.dylib (compatibility version 1.0.0, current version 1.1.0, weak)
❯ otool -l swift-format | rg -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 32
         path /usr/lib/swift (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path @loader_path (offset 12)
--
          cmd LC_RPATH
      cmdsize 112
         path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx (offset 12)
```

**Comment by Bo98** [comment]
The general mixup of versions seen in the other PRs is possibly the cause. Perhaps a more verbose build in the other PRs might reveal more of what it's doing and the commands it's running.

**Comment by Bo98** [comment]
Oh I'm being silly. Of course it doesn't work - `/usr/local/opt/swift/Swift-5.4.xctoolchain/usr/bin` won't be in the PATH.

**Comment by carlocab** [comment]
Let's try manually adding symlinks here to see if it's worth it.

File: Formula/swift-format.rb [hunk 1]
```diff
@@ -22,10 +22,21 @@ class SwiftFormat < Formula
...
+    depends_on xcode: ["12.5", :build]
+  else
+    depends_on "swift"
+  end
 
   def install
-    system "swift", "build", "--disable-sandbox", "-c", "release"
+    swift = if OS.mac? && MacOS.version <= :catalina
+      f = Formula["swift"]
+      f.opt_prefix/"Swift-#{f.version.major_minor}.xctoolchain/usr/bin/swift"
```
Comment (Bo98) [inline Formula/swift-format.rb:34]: Sorry this should be:

```suggestion
      f.opt_prefix/"Swift-#{f.version}.xctoolchain/usr/bin/swift"
```

**Comment by carlocab** [comment]
Oh hey, that worked. I think Linux just needs to be built with Clang.

**Pull request closed without merging**